### PR TITLE
Tidying

### PR DESCRIPTION
**Pull request by mbway**
I have made several small changes:
- ran `cargo fmt` and `cargo clippy`
- added more information to some documentation comments
- removed an empty module
- added formatting and linting checks to the CI
    - I have not used Travis CI before so there may be problems with my changes
    - I added FreeBSD as part of the CI process. I see from #65  that this will fail currently so the configuration currently allows that.
    - I see that the latest commit is failing on rust nightly. The CI configuration allows build failures on nightly now which is probably advisable
    - `travis-cargo` has not been updated since 2016 so I'm not sure whether it still works, but I left it in anyway
    - it should be possible to automatically push to crates.io by pushing to the `release` branch but I don't have a crates.io account so can't experiment with that.
    - I made some changes that were recommended by travis when I ran the travis job on another repo for testing, namely:
```
root: deprecated key sudo (The key `sudo` has no effect anymore.)
root: missing dist, using the default xenial
root: missing os, using the default linux
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -29,12 +44,14 @@ will appear if the kernel ring buffer is overrun before messages are read. I try
 match [libevdev](https://www.freedesktop.org/software/libevdev/doc/latest/)
 closely, where possible.
 
-Writing to devices is not yet supported (eg, turning LEDs on).
```
Comment (mbway) [inline README.md:32]: I removed this comment because the documentation indicates this is now supported. Is this correct?

Comment (coolreader18) [reply README.md:32]: I think so, I'm still not actually sure what the "turning LEDS on" thing is but yes, stuff like uinputs can be written to.

File: src/constants.rs [hunk 2]
```diff
@@ -1,5 +1,7 @@
 /// Event types supported by the device.
 ///
+/// Values correspond to [/usr/include/linux/input-event-codes.h](https://raw.githubusercontent.com/torvalds/linux/master/include/uapi/linux/input-event-codes.h)
```
Comment (coolreader18) [inline src/constants.rs:3]: This could probably just be a normal `github.com/` link

File: src/scancodes.rs [hunk 3]
```diff
@@ -1,5 +1,7 @@
 /// Scancodes for key presses.
 ///
+/// Values correspond to [/usr/include/linux/input-event-codes.h](https://raw.githubusercontent.com/torvalds/linux/master/include/uapi/linux/input-event-codes.h)
```
Comment (coolreader18) [inline src/scancodes.rs:3]: Same here

File: src/tokio_stream.rs [hunk 4]
```diff
@@ -1 +0,0 @@
-
```
Comment (coolreader18) [inline src/tokio_stream.rs:1]: Oh, hah, thanks.

**Review by coolreader18 — APPROVED** [review]
LGTM, thanks for contributing!

**Comment by coolreader18** [comment]
I'll probably switch to gh actions at some point, but thanks for tidying up the ci script.

**Pull request merged**